[bug]: Missing peer dependencies when installing `@magicui/highlighter` with shadcn monorepo

### Describe the bug

## Description

When using `shadcn/ui` in a monorepo setup, installing the `@magicui/highlighter` component via CLI does not install required third-party dependencies. This results in build errors due to missing modules.

## Expected Behavior

All required dependencies (including third-party packages) should be installed automatically, or clearly documented.

## Actual Behavior

The following errors occur due to missing dependencies:

```ts
import { useInView } from "motion/react"
// ❌ Cannot find module 'motion/react'

import { type RoughAnnotation } from "rough-notation/lib/model"
// ❌ Cannot find module 'rough-notation/lib/model'
```

## Missing Dependencies

* `motion` (for `motion/react`)
* `rough-notation/lib/model`


## Additional Context**

* The issue occurs in a monorepo setup (`packages/ui`, `apps/web`)
* The CLI successfully adds component files but does not handle peer dependencies

## Suggested Fix

1. Ensure `shadcn` CLI installs required dependencies automatically when adding components
2. OR clearly list required dependencies during installation output
3. OR update documentation to include manual installation steps

## Documentation Issue

The current docs for the Highlighter component are incomplete:
👉 [https://magicui.design/docs/components/highlighter](https://magicui.design/docs/components/highlighter)

* The **manual installation section does not mention required third-party dependencies**
* It should include:

```bash
pnpm add motion rough-notation
```

## Environment

* Package manager: `pnpm`
* Setup: Monorepo (`apps/web`, `packages/ui`)
* Command used: `pnpm dlx shadcn@latest add @magicui/highlighter -c apps/web`



### Affected component/components

https://magicui.design/docs/components/highlighter

### How to reproduce

1. Initialize a `shadcn/ui` monorepo project
2. Run the following command:

```bash
pnpm dlx shadcn@latest add @magicui/highlighter -c apps/web
```

3. Observe installation output:

```
✔ Checking registry.
✔ Installing components.
✔ Created 1 file:
  - packages/ui/src/components/highlighter.tsx
ℹ Updated 1 file:
  - packages/ui/src/styles/globals.css
```

4. Start the project or type-check

### Codesandbox/StackBlitz link

_No response_

### Logs

```bash

```

### System Info

```bash
Version: 1.115.0
Commit: 41dd792b5e652393e7787322889ed5fdc58bd75b
Date: 2026-04-07T22:04:57Z
Electron: 39.8.5
ElectronBuildId: 13703022
Chromium: 142.0.7444.265
Node.js: 22.22.1
V8: 14.2.231.22-electron.0
OS: Darwin arm64 25.3.0
```

### Before submitting

- [x] I've made research efforts and searched the documentation
- [x] I've searched for existing issues

--- a/apps/www/content/docs/components/highlighter.mdx
+++ b/apps/www/content/docs/components/highlighter.mdx
@@ -28,6 +28,12 @@ npx shadcn@latest add @magicui/highlighter
 
 <Steps>
 
+<Step>Install the following dependencies:</Step>
+
+```bash
+pnpm add motion rough-notation
+```
+
 <Step>Copy and paste the following code into your project.</Step>
 
 <ComponentSource name="highlighter" />

--- a/apps/www/public/r/highlighter-demo.json
+++ b/apps/www/public/r/highlighter-demo.json
@@ -5,6 +5,7 @@
   "title": "Highlighter Demo",
   "description": "Example showing the demo of a Highlighter",
   "dependencies": [
+    "motion",
     "rough-notation"
   ],
   "registryDependencies": [

--- a/apps/www/public/r/highlighter.json
+++ b/apps/www/public/r/highlighter.json
@@ -5,6 +5,7 @@
   "title": "Highlighter",
   "description": "A text highlighter that mimics the effect of a human-drawn marker stroke.",
   "dependencies": [
+    "motion",
     "rough-notation"
   ],
   "files": [

--- a/apps/www/public/r/registry.json
+++ b/apps/www/public/r/registry.json
@@ -1219,6 +1219,7 @@
       "title": "Highlighter",
       "description": "A text highlighter that mimics the effect of a human-drawn marker stroke.",
       "dependencies": [
+        "motion",
         "rough-notation"
       ],
       "files": [
@@ -3449,6 +3450,7 @@
       "title": "Highlighter Demo",
       "description": "Example showing the demo of a Highlighter",
       "dependencies": [
+        "motion",
         "rough-notation"
       ],
       "registryDependencies": [

--- a/apps/www/public/registry.json
+++ b/apps/www/public/registry.json
@@ -1219,6 +1219,7 @@
       "title": "Highlighter",
       "description": "A text highlighter that mimics the effect of a human-drawn marker stroke.",
       "dependencies": [
+        "motion",
         "rough-notation"
       ],
       "files": [
@@ -3449,6 +3450,7 @@
       "title": "Highlighter Demo",
       "description": "Example showing the demo of a Highlighter",
       "dependencies": [
+        "motion",
         "rough-notation"
       ],
       "registryDependencies": [

--- a/apps/www/registry.json
+++ b/apps/www/registry.json
@@ -1219,6 +1219,7 @@
       "title": "Highlighter",
       "description": "A text highlighter that mimics the effect of a human-drawn marker stroke.",
       "dependencies": [
+        "motion",
         "rough-notation"
       ],
       "files": [
@@ -3449,6 +3450,7 @@
       "title": "Highlighter Demo",
       "description": "Example showing the demo of a Highlighter",
       "dependencies": [
+        "motion",
         "rough-notation"
       ],
       "registryDependencies": [

--- a/apps/www/registry/registry-examples.ts
+++ b/apps/www/registry/registry-examples.ts
@@ -1879,7 +1879,7 @@ export const examples: Registry["items"] = [
         type: "registry:ui",
       },
     ],
-    dependencies: ["rough-notation"],
+    dependencies: ["motion", "rough-notation"],
   },
   {
     name: "animated-theme-toggler-demo",

--- a/apps/www/registry/registry-ui.ts
+++ b/apps/www/registry/registry-ui.ts
@@ -1182,7 +1182,7 @@ export const ui: Registry["items"] = [
         type: "registry:ui",
       },
     ],
-    dependencies: ["rough-notation"],
+    dependencies: ["motion", "rough-notation"],
   },
   {
     name: "animated-theme-toggler",

